crates/daphne-server/src/router/mod.rs

```diff
@@ -369,7 +371,12 @@ where
                 .server_metrics()
                 .auth_method_inc(metrics::AuthMethod::BearerToken);
             state
-                .check_bearer_token(&token, sender, request.task_id)
+                .check_bearer_token(
+                    &token,
+                    sender,
+                    request.task_id.unwrap(), // task ids are mandatory
+                    request.taskprov.is_some(),
```

Comment:
Is this enough to determine if the task is using taskprov? It's different from how we used to test this

Reply:
We should also consult the `DapGlobalConfig` to see if we've enabled taskprov support (`allow_taskprov`). If we haven't, then the advertisement in the request will be ignored.

Reply:
I'm doing that inside the `check_bearer_token` method

Reply:
Sounds good. Maybe make a note here if you think it's not clear from the control flow. (I haven't looked.)